[Bug]: Test `gradient_factory::convert_from_rgb` fails on i686

### What happened?

I am trying to build conky on i686 with glibc. Test `gradient_factory::convert_from_rgb` fails. See more at <https://github.com/void-linux/void-packages/actions/runs/4469212726/jobs/7851003126?pr=42879>.

### Version

1.18.3

### Which OS/distro are you seeing the problem on?

Linux (other)

### Conky config

_No response_

### Stack trace

_No response_

### Relevant log output

```Shell
[0/1] Running tests...
Test project /builddir/conky-1.18.3/build
      Start  1: to_real_path becomes homedir
 1/16 Test  #1: to_real_path becomes homedir ................................................   Passed    0.01 sec
      Start  2: environment variables are substituted correctly
 2/16 Test  #2: environment variables are substituted correctly .............................   Passed    0.01 sec
      Start  3: cpu_percentage and cpu_barval return correct values
 3/16 Test  #3: cpu_percentage and cpu_barval return correct values .........................   Passed    0.01 sec
      Start  4: mem_percentage and mem_barval return correct values
 4/16 Test  #4: mem_percentage and mem_barval return correct values .........................   Passed    0.01 sec
      Start  5: mem_with_buffers_barval returns correct value
 5/16 Test  #5: mem_with_buffers_barval returns correct value ...............................   Passed    0.01 sec
      Start  6: swap_percentage and swap_barval return correct values
 6/16 Test  #6: swap_percentage and swap_barval return correct values .......................   Passed    0.01 sec
      Start  7: Expressions can be evaluated
 7/16 Test  #7: Expressions can be evaluated ................................................   Passed    0.01 sec
      Start  8: get_entropy_avail returns 0
 8/16 Test  #8: get_entropy_avail returns 0 .................................................   Passed    0.01 sec
      Start  9: remove_comments returns correct value
 9/16 Test  #9: remove_comments returns correct value .......................................   Passed    0.01 sec
      Start 10: diskiographval returns correct value
10/16 Test #10: diskiographval returns correct value ........................................   Passed    0.01 sec
      Start 11: fs_free_percentage returns correct value
11/16 Test #11: fs_free_percentage returns correct value ....................................   Passed    0.01 sec
      Start 12: gradient_factory::convert_from_rgb returns correct value
12/16 Test #12: gradient_factory::convert_from_rgb returns correct value ....................***Failed    0.01 sec
Filters: gradient_factory::convert_from_rgb returns correct value

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test-conky is a Catch v2.13.9 host application.
Run with -? for options

-------------------------------------------------------------------------------
gradient_factory::convert_from_rgb returns correct value
  hsv_gradient_factory
  saturation
-------------------------------------------------------------------------------
/builddir/conky-1.18.3/tests/test-gradient.cc:72
...............................................................................

/builddir/conky-1.18.3/tests/test-gradient.cc:72: FAILED:
  REQUIRE( result[1] == expected_saturation )
with expansion:
  13224 (0x33a8) == 122880 (0x1e000)

-------------------------------------------------------------------------------
gradient_factory::convert_from_rgb returns correct value
  hcl_gradient_factory
  luma
-------------------------------------------------------------------------------
/builddir/conky-1.18.3/tests/test-gradient.cc:86
...............................................................................

/builddir/conky-1.18.3/tests/test-gradient.cc:86: FAILED:
  REQUIRE( result[2] == expected_luma )
with expansion:
  96822 (0x17a36) == 20712665 (0x13c0cd9)

===============================================================================
test cases: 1 | 1 failed
assertions: 9 | 7 passed | 2 failed


      Start 13: gradient_factory should convert to and from rgb and get the initial value
13/16 Test #13: gradient_factory should convert to and from rgb and get the initial value ...***Failed    0.01 sec
Filters: gradient_factory should convert to and from rgb and get the initial value

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test-conky is a Catch v2.13.9 host application.
Run with -? for options

-------------------------------------------------------------------------------
gradient_factory should convert to and from rgb and get the initial value
  rgb_gradient_factory
-------------------------------------------------------------------------------
/builddir/conky-1.18.3/tests/test-gradient.cc:95
...............................................................................

/builddir/conky-1.18.3/tests/test-gradient.cc:101: FAILED:
  REQUIRE( result == colour )
with expansion:
  {?} == {?}

-------------------------------------------------------------------------------
gradient_factory should convert to and from rgb and get the initial value
  hsv_gradient_factory
-------------------------------------------------------------------------------
/builddir/conky-1.18.3/tests/test-gradient.cc:106
...............................................................................

/builddir/conky-1.18.3/tests/test-gradient.cc:112: FAILED:
  REQUIRE( result == colour )
with expansion:
  {?} == {?}

-------------------------------------------------------------------------------
gradient_factory should convert to and from rgb and get the initial value
  hcl_gradient_factory
-------------------------------------------------------------------------------
/builddir/conky-1.18.3/tests/test-gradient.cc:117
...............................................................................

/builddir/conky-1.18.3/tests/test-gradient.cc:123: FAILED:
  REQUIRE( result == colour )
with expansion:
  {?} == {?}

===============================================================================
test cases: 1 | 1 failed
assertions: 3 | 3 failed


      Start 14: scan_graph correctly parses input strings
14/16 Test #14: scan_graph correctly parses input strings ...................................   Passed    0.01 sec
      Start 15: scan_command correctly parses input strings
15/16 Test #15: scan_command correctly parses input strings .................................   Passed    0.01 sec
      Start 16: parse_color correctly parses colours
16/16 Test #16: parse_color correctly parses colours ........................................   Passed    0.01 sec

88% tests passed, 2 tests failed out of 16

Total Test time (real) =   0.14 sec

The following tests FAILED:
	 12 - gradient_factory::convert_from_rgb returns correct value (Failed)
	 13 - gradient_factory should convert to and from rgb and get the initial value (Failed)
Errors while running CTest
FAILED: CMakeFiles/test.util 
cd /builddir/conky-1.18.3/build && /usr/bin/ctest --force-new-ctest-process
ninja: build stopped: subcommand failed.
```

--- a/src/gradient.cc
+++ b/src/gradient.cc
@@ -199,7 +199,7 @@ namespace {
 // Using Rec.2020 color space
 // Y' = 0.2627 x R + 0.6780 x G + 0.0593 x B
 long get_luma(long *const rgb) {
-  return 360L * (2627L * rgb[0] + 6780L * rgb[1] + 593L * rgb[2]) / 10000L;
+  return 360L * (uint64_t)(2627L * rgb[0] + 6780L * rgb[1] + 593L * rgb[2]) / 10000L;
 }
 
 // Using Rec.2020 color space

--- a/src/gradient.cc
+++ b/src/gradient.cc
@@ -156,7 +156,7 @@ void hsv_gradient_factory::convert_from_scaled_rgb(long *const scaled,
   auto value = get_value(scaled);
   auto minimum = get_minimum(scaled);
   auto chroma = value - minimum;
-  auto saturation = (SCALE360 * chroma) / value;
+  long saturation = (SCALE360 * (uint64_t)chroma) / value;
 
   target[0] = get_hue(scaled, chroma, value);
   target[1] = saturation;

